Common/Core/RecoDecay.h

```diff
@@ -644,7 +645,10 @@ class RecoDecay
       // for (int i = 0; i < stage; i++) // Indent to make the tree look nice.
       //   printf(" ");
       // printf("Stage %d: Adding %d (PDG %d) as final daughter.\n", stage, index, PDGParticle);
-      list->push_back(particle.globalIndex());
+      // check if the particle comes from a decay process
+      if (particle.getProcess() == TMCProcess::kPDecay) {
+        list->push_back(particle.globalIndex());
```

Comment:
There should be no conditions in this block. The decision to push the particle is made by the value of `isFinal`.

Reply:
I have to add the particle to the list only if it comes from a decay, so I would simply move this condition. Where do you want me to put it?